project_forecast_line_priority/models/project_task.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 Therp BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from datetime import timedelta
+
+from odoo import fields, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    def write(self, vals):
+        """Set forecast_date_planned_end based on priority"""
+        # If user sets end date manually
+        if "forecast_date_planned_end" in vals or "date_deadline" in vals:
+            return super().write(vals)
+        if "priority" not in vals:
+            return super().write(vals)
+        priority = vals.get("priority", -5)
+        for this in self:
+            # date deadline is set, so ignore this one
+            if this.date_deadline:
+                continue
+            # if priority is not set, do nothing
+            if int(priority) < 0:
```

Comment:
less then, or less then equal `<=` as a task with priority zero has basically no priority, and zero is the default?

Reply:
Actually priority 0 means no priority, but the module should also set forecast_date_planned_end in this case too

Reply:
@NL66278 functionally what @ntsirintanis says here is correct, the module _can_ set a planned end date for `priority` 0 as well so we should do only nothing if even that is not the case.

Reply:
Actually there is a corner case that priority field can be false. This happens when a portal user is creating the task; the priority field accepts falsy values on that case, i.e., its default value is False.